src/avram/migrator/runner.cr

```diff
@@ -95,6 +95,44 @@ class Avram::Migrator::Runner
     end
   end
 
+  def self.restore_db(restore_file : String, quiet : Bool = false)
+    if File.exists?(restore_file)
+      run "psql -q #{cmd_args} -v ON_ERROR_STOP=1 < #{restore_file}"
+      unless quiet
+        puts "Done restoring #{db_name.colorize(:green)}"
+      end
+    else
+      raise "Unable to locate the restore file: #{restore_file}"
+    end
+  end
+
+  def self.dump_db(quiet : Bool = false)
+    run "pg_dump -U #{db_user} -h #{db_host} -p #{db_port} -s #{db_name} > lucky_#{db_name}_dump-#{Time.now.to_s("%Y%m%d%H%I")}.sql"
+    unless quiet
+      puts "Done dumping #{db_name.colorize(:green)}"
+    end
+  rescue e : Exception
+    message = e.message.to_s
+    if message.includes?("does not exist")
+      raise <<-ERROR
+      The database #{db_name} does not exist on host #{db_host}.
+
+      Try running 'lucky db.create' first.
+      ERROR
+    elsif message.includes?("Connection refused")
+      raise <<-ERROR
+      Unable to connect to db #{db_name}.
+
+      Try this...
+
+        ▸ Check your settings in 'config/database.cr'.
+        ▸ Run 'luck db.verify_connection' to ensure you can connect
+      ERROR
```

Comment:
Maybe run the verifiy connection task before dumping so you don't need these. I think that task handles most of this